src/misc.jl

```diff
@@ -104,6 +133,18 @@ findat(a::AbstractArray, b::AbstractArray) = findat!(Array(Int, size(b)), a, b)
 # c: categories
 # k: the maximum integer in x
 
+"""
+    indicatormat(x::IntegerArray, k::Integer; sparse::Bool=false)
+
+### Args:
+* `x`: A, `IntegerArray`.
+* `k`: An `Integer`
+* `sparse`: A keyword argument, of `Bool` type.
+
+Construct a boolean matrix `r` of size `(k, length(x))` such that `r[x[i], i] = true` and all other elements are set to false.
```

Comment:
Could be useful to give a literary explanation. Something like this maybe?

Construct a boolean matrix with `k` rows and one column for each element in `x`, such that in each column the row index of the `true` element is given by the corresponding element from `x`. That is, the resulting matrix  `r` is such that `r[x[i], i] = true` and all other elements are `false`.